### DISCUSSION TITLE
/run/containerd/containerd.sock: No such file or directory

### DISCUSSION THREAD

**Discussion by davidhsingyuchen** [containerd/nerdctl#1575] [Q&A] [answered]
Environment: I'm using the default lima instance without any customization. Shell into the VM via `limactl shell default`.

According to the [FAQ](https://github.com/containerd/nerdctl/blob/main/docs/faq.md#containerd-socket-runcontainerdcontainerdsock), the containerd address is `/run/containerd/containerd.sock`, and `nerdctl` does work by connecting to that address, but I don't know why `ls` says `No such file or directory`:

```sh
davidhyc@lima-default:/Users/davidhyc$ nerdctl --address /run/containerd/containerd.sock images
REPOSITORY    TAG       IMAGE ID        CREATED          PLATFORM          SIZE       BLOB SIZE
alpine        latest    8914eb54f968    2 minutes ago    linux/arm64/v8    7.8 MiB    3.1 MiB
davidhyc@lima-default:/Users/davidhyc$ ls /run/containerd/containerd.sock
ls: cannot access '/run/containerd/containerd.sock': No such file or directory
```

I thought it may be due to directory permission, but I can ls `/run`, and there does not seem to be a `containerd` dir:

```sh
davidhyc@lima-default:/Users/davidhyc$ ls /run/containerd
ls: cannot access '/run/containerd': No such file or directory
davidhyc@lima-default:/Users/davidhyc$ ls /run
agetty.reload  credentials   dbus             initctl         lima-guestagent.sock  lvm           multipathd.pid  screen           snapd-snap.socket  sudo        user
blkid          crond.pid     dmeventd-client  initramfs       lima-ssh-ready        machine-id    needrestart     sendsigs.omit.d  snapd.socket       systemd     utmp
cloud-init     crond.reboot  dmeventd-server  irqbalance      lock                  motd.dynamic  netns           shm              sshd               tmpfiles.d  uuidd
console-setup  cryptsetup    fsck             lima-boot-done  log                   mount         network         snapd            sshd.pid           udev
```

I've searched in both lima and nerdctl but no luck. Appreciate any guidance, thanks!

**Comment by AkihiroSuda** [comment, marked as answer]
nerdctl uses `/run/containerd/containerd.sock` that exists in the daemon's mount namespace (`/proc/<PID of containerd>/root/run/containerd/containerd.sock`)

**Reply by davidhsingyuchen** [reply]
Thank you. Created https://github.com/containerd/nerdctl/pull/1590 to add this to FAQ.
